src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -2,13 +1,44 @@
+import React, { useCallback, useState } from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setQuery, setStatus } from '../../features/filter';
+import { Status } from '../../types/Status';
+import debounce from 'lodash.debounce';
 
 export const TodoFilter: React.FC = () => {
+  const [localQuery, setLocalQuery] = useState('');
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const dispatch = useAppDispatch();
+
+  const applyQuery = debounce((string: string) => {
+    dispatch(setQuery(string));
+  }, 300);
+
+  const handleCloseButton = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault();
+      setLocalQuery('');
+      applyQuery('');
+    },
+    [applyQuery],
+  );
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    applyQuery(event.target.value);
+    setLocalQuery(event.target.value);
+  };
+
+  const handleStatus = (event: React.ChangeEvent) => {
```

Comment:
The `handleStatus` function uses `HTMLOptionElement` for type assertion, which is incorrect. It should be `HTMLSelectElement` since the event target is a select element. Change `HTMLOptionElement` to `HTMLSelectElement`.